### DISCUSSION TITLE
Add support for setting Media Injection files using gmtool/Java API

### DISCUSSION THREAD

**Discussion by bhein-sherpa6** [Genymobile/feedback#66] [Genymotion Desktop] [answered]
I have a requirement to scan a QR code from an app, and to do so we currently have to save the QR to a .png file, set it using the Media Injection widget in the emulator toolbar, at which point the image appears by default when the camera is opened.

We would like to be able to remove this manual step of using the Media Injection widget. Is it possible to do this either through:
a. gmtool
b. Java API

If this is currently already possible, my apologies, but I've looked through the documentation of both of these and nothing regarding setting the default image when the camera is opened programmatically.

Thank you far in advance. We are truly finding Genymotion extremely useful to our current business needs.

Cheers!

Brandon

**Comment by bhein-sherpa6** [comment]
Any feedback on this one? We'd like to get the ball rolling sooner than later.

From my understanding you are a small team, so anything I can do on my side to move things along?

**Comment by bhein-sherpa6** [comment]
@agavignet I apologize for tagging directly, but there doesn't appear to be any movement on this topic or any other in this forum for that matter, so I picked a name. Is there somewhere else that I need to issue this request? We really enjoy your solution but if you don't have the manpower to address these requests we need to know so we can plan an alternative approach.

**Comment by agavignet** [comment, marked as answer]
Hi @bhein-sherpa6,

I'm truly sorry for the response delay, I did not catch the notification. Fortunately our support team pointed it to me. Regarding your need, it makes perfect sense to me, the team will think about a way to inject an image via cmd line, I'll let you know. Unfortunately for now I have no workaround for you :/

Adrien

**Reply by bhein-sherpa6** [reply]
Thank you for your response. I look forward to this feature at some point in the future.

**Reply by agavignet** [reply]
@bhein-sherpa6 Would it be OK for you if  by default (and as a first step), Genymotion would use `default_front.png` and `default_back.png` located in the media folder (if present), for both back and front Android cameras by default?

**Reply by bhein-sherpa6** [reply]
@agavignet Would that be a replacement of the current behavior, which is to default to the "Dummy Camera"?:

<img width="887" alt="image" src="https://github.com/Genymobile/feedback/assets/122338660/4d54f727-0c1a-48e2-9539-17f9e299a1ba">

I'm not sure if/how that will impact other users. Assuming that if you made this change, then to programmatically ensure that the camera opened up to my custom image, I'd need to name the image "default_front.png" for it to be the default? I'd also need to manage _where_ that media folder exists, depending on the execution environment (not ideal).

This would at least be a good workaround, but ultimately I was imagining a solution where I could just call a gmtool/API function with a file and it would set that image for me. This way I wouldn't need to care about the media folder location, which can change depending on the environment. But as you said, your proposed solution would definitely be a good first step.

**Comment by agavignet** [comment]
Hi @bhein-sherpa6, Desktop 3.6 has been released, it does not cover entirely your need but it's going in the right direction, in this first iteration inputs camera for each device are persistent after a virtual device restart, so you can set and forget.

Adrien

**Reply by bhein-sherpa6** [reply]
Thank you, this is a necessary piece. But as you've stated does not cover the initial need to be able to set this image programmatically to start with (instead of using the Media Injection widget manually). I look forward to your continued improvements. As I've stated a few times and in other posts, we are truly finding a ton of value from your solution and are pleased with the progress. Thank you guys!
